test-data/genops-statements.test

```diff
@@ -170,3 +170,32 @@ L5:
 L6:
     r2 = None
     return r2
+
+[case testForList]
+from typing import List
+
+def f(ls: List[int]) -> int:
+    y = 0
+    for x in ls:
+        y = y + x
+    return y
+[out]
+def f(ls):
+    ls :: list
+    y, r0, r1, x, r2, r3 :: int
+L0:
+    y = 0
+    r0 = 0
+    r1 = 1
+    r2 = len ls :: list
```

Comment:
For compatibility with Python semantics, we unfortunately need to re-evaluate the length of the list on each iteration, since the loop body might mutate the list. In the future we can implement loop-invariant code motion to optimize this, but that can wait.